### DISCUSSION TITLE
Capture first-time sign-in with oidc flow.

### DISCUSSION THREAD

**Discussion by shreeharsha-factly** [ory/kratos#3783] [Q&A]
If the user logs in or registers with OIDC flow. Is there a way to capture first-time sign-in with the help of hooks or any method? 

If the user is using email-password flow then to capture first-time sign-in we are relying on after verification hook. But with OIDC flow after verification hook is not triggering. So how can we handle this use-case with OIDC flow?

**Comment by vinckr** [comment]
Hello @shreeharsha-factly 

In this case you should be able to use a "after oidc login" Ory Action.
